packages/aws-cdk/test/api/cloudformation-deployments.test.ts

```diff
@@ -142,3 +161,751 @@ test('if toolkit stack cannot be found but SSM parameter name is present deploym
 
   expect(requestedParameterName!).toEqual('/some/parameter');
 });
+
+test('readCurrentTemplateWithNestedStacks() can handle non-Resources in the template', async () => {
+  const cfnStack = instanceMockFrom((CloudFormationStack as any));
```

Comment:
We have a `FakeCloudFormationStack` class defined [here](https://github.com/aws/aws-cdk/blob/554645d1ca4d4223a0ab9490ea5cf5c0f201ed3a/packages/aws-cdk/test/api/fake-cloudformation-stack.ts#L10-L10) - you might want to use it instead.

Same comment everywhere below.